File: exercises/RB101_small_problems/easy_1/bannerizer.rb
def print_in_box(str)
  horizontal_line = "+#{'-' * (str.size + 2)}+"
  empty_line = "|#{' ' * (str.size + 2)}|"

  puts horizontal_line
  puts empty_line
  puts "| #{str} |"
  puts empty_line
  puts horizontal_line
end

print_in_box('To boldly go where no one has gone before.')

File: exercises/JS210_Small_Problems/Easy_2/bannerizer.js
function logInBox(text) {
  const horizontalRule = `+${repeatedChar('-', text.length + 2)}+`;
  const emptyLine = `|${repeatedChar(' ', text.length + 2)}|`;

  console.log(horizontalRule);
  console.log(emptyLine);
  console.log(`| ${text} |`);
  console.log(emptyLine);
  console.log(horizontalRule)
}

function repeatedChar(char, times) {
  let repeated = '';

  while (repeated.length < times) {
    repeated += char;
  }

  return repeated;
}

logInBox('To boldly go where no one has gone before.');

// +--------------------------------------------+
// |                                            |
// | To boldly go where no one has gone before. |
// |                                            |
// +--------------------------------------------+